cf-agent/cf-agent.c

```diff
@@ -238,11 +238,21 @@ int main(int argc, char *argv[])
     {
         Log(LOG_LEVEL_ERR, "CFEngine was not able to get confirmation of promises from cf-promises, so going to failsafe");
         EvalContextClassPutHard(ctx, "failsafe_fallback", "attribute_name=Errors,source=agent");
-        GenericAgentConfigSetInputFile(config, GetInputDir(), "failsafe.cf");
-        policy = LoadPolicy(ctx, config);
+        
+        if (CheckAndGenerateFailsafe(GetInputDir(), "failsafe.cf"))
+        {
+            GenericAgentConfigSetInputFile(config, GetInputDir(), "failsafe.cf");
+            policy = LoadPolicy(ctx, config);
+        }
     }
     assert(policy);
-
+    
+    if (!policy)
```

Comment:
So assert() should be redundant as well. I think it is possible to return NULL by LoadPolicy() anyway but let's not argue. I will leave assert() as was before.